app/src/main/java/net/gsantner/markor/format/zimwiki/ZimWikiTextActions.java

```diff
@@ -0,0 +1,257 @@
+                    SearchOrCustomTextDialogCreator.showMarkdownHeadlineDialog(_activity, origText, (text, lineNr) -> {
+                        _hlEditor.setSelection(StringUtils.getIndexFromLineOffset(origText, lineNr, 0));
+                    });
+                    return true;
+                }
+                case R.string.tmaid_common_indent:
+                    runRegexReplaceAction(ZimWikiReplacePatternGenerator.indentOneTab());
+                    return true;
+                case R.string.tmaid_common_deindent: {
+                    runRegexReplaceAction(ZimWikiReplacePatternGenerator.deindentOneTab());
+                    runRenumberOrderedListIfRequired();
+                    return true;
+                }
+                default: {
+                    return runCommonTextAction(_context.getString(_action));
+                }
+            }
+        }
+
+        @Override
+        public boolean onLongClick(View view) {
+            switch (_action) {
+                case R.string.tmaid_common_open_link_browser: {
+                    new CommonTextActions(_activity, _hlEditor).runAction(CommonTextActions.ACTION_SEARCH);
+                    return true;
+                }
+                case R.string.tmaid_common_special_key: {
+                    new CommonTextActions(_activity, _hlEditor).runAction(CommonTextActions.ACTION_JUMP_BOTTOM_TOP);
+                    return true;
+                }
+                case R.string.tmaid_zimwiki_insert_image: {
+                    // TODO: adapt to zim wiki
+                    int pos = _hlEditor.getSelectionStart();
+                    _hlEditor.getText().insert(pos, "<img style=\"width:auto;max-height: 256px;\" src=\"\" />");
+                    _hlEditor.setSelection(pos + 48);
+                    return true;
+                }
+                case R.string.tmaid_common_time: {
+                    runCommonTextAction("tmaid_common_time_insert_timestamp");
+                    return true;
+                }
+                case R.string.tmaid_zimwiki_code_inline: {
+                    _hlEditor.disableHighlighterAutoFormat();
+                    final int c = _hlEditor.setSelectionExpandWholeLines();
+                    _hlEditor.getText().insert(_hlEditor.getSelectionStart(), "\n'''\n");
+                    _hlEditor.getText().insert(_hlEditor.getSelectionEnd(), "\n'''\n");
+                    _hlEditor.setSelection(c + "\n'''\n".length());
+                    _hlEditor.enableHighlighterAutoFormat();
+                    Toast.makeText(_activity, R.string.code_block, Toast.LENGTH_SHORT).show();
+                    return true;
+                }
+                case R.string.tmaid_common_ordered_list_number: {
+                    // TODO: adapt to zim wiki
+                    AutoFormatter.renumberOrderedList(_hlEditor.getText(), StringUtils.getSelection(_hlEditor)[0], ZimWikiAutoFormat.getPrefixPatterns());
+                }
```

Comment:
Please see the code recently added here: https://github.com/gsantner/markor/pull/1129/files#diff-bbc685651c09c955da0b077fe835a688483119842448adedfa7296e0a8920b68R256-R262.

You will want to include this here

Reply:
@fredericjacob ^^

Reply:
Does not work yet currently because of zim using tabs instead of spaces, but I added this to my TODO backlog. :+1: